gabbi/case.py

```diff
@@ -187,13 +190,27 @@ def _clean_query_value(self, value):
         return value
 
     def _environ_replace(self, message):
-        """Replace an indicator in a message with the environment value."""
+        """Replace an indicator in a message with the environment value.
+
+        If value can be a number, cast it as such. If value is a form of
+        "null", "true", or "false" cast it to None, True, False.
+        """
         value = re.sub(self._replacer_regex('ENVIRON'),
                        self._environ_replacer, message)
-        if value == "False":
+        try:
+            if '.' in value:
+                value = float(value)
+            else:
+                value = int(value)
+            return value
+        except ValueError:
+            pass
+        if value.lower() == "false":
             return False
-        if value == "True":
+        if value.lower() == "true":
             return True
+        if value.lower() == "null":
+            return None
```

Comment:
this is the value of the environment variable being parsed and all environment variables are strings but in this case we want them have stronger types than that so they can actually be used as those types in things like json.

and yes, caching `value.lower()` is silly. not completely silly, but silly